### ISSUE TITLE
[libproxy] Dependency in statically compiled version

### ISSUE THREAD

**Issue by illera88** [microsoft/vcpkg#28417]
**Describe the bug**
Libproxy depends on `libmodman` which is a [static target](https://github.com/libproxy/libproxy/blob/8fec01ed4b95afc71bf7710bf5b736a5de03b343/libmodman/CMakeLists.txt#L10). The problem is that then compiling statically you need `libmodman.lib` too or you will get linking errors when using `libproxy` in your code.
Should I move `libmodman.lib` to the `libs` folder when installing `libproxy`?

**To Reproduce**
1. ./vcpkg install libproxy:x64-windows-static
2. Try to use libproxy in your `main.cpp`

**Expected behavior**
It compiles and links fine. Instead there are linking errors:

```
3>libproxy.lib(proxy.cpp.obj) : error LNK2019: unresolved external symbol "public: __cdecl libmodman::module_manager::~module_manager(void)" (??1module_manager@libmodman@@QEAA@XZ) referenced in function "public: __cdecl libproxy::proxy_factory::~proxy_factory(void)" (??1proxy_factory@libproxy@@QEAA@XZ)
3>libproxy.lib(proxy.cpp.obj) : error LNK2019: unresolved external symbol "public: bool __cdecl libmodman::module_manager::load_builtin(struct mm_module *)" (?load_builtin@module_manager@libmodman@@QEAA_NPEAUmm_module@@@Z) referenced in function "public: __cdecl libproxy::proxy_factory::proxy_factory(void)" (??0proxy_factory@libproxy@@QEAA@XZ)
3>libproxy.lib(proxy.cpp.obj) : error LNK2019: unresolved external symbol "public: bool __cdecl libmodman::module_manager::load_dir(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,bool)" (?load_dir@module_manager@libmodman@@QEAA_NAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@_N@Z) referenced in function "public: __cdecl libproxy::proxy_factory::proxy_factory(void)" (??0proxy_factory@libproxy@@QEAA@XZ)

```

## How should I resolve this?
Should I just copy `libmodman.lib` to the `libs` folder?

**Comment by xiaozhuai** [comment]
@illera88 Should be fixed by #28457

**Comment by xiaozhuai** [comment]
Please reopen this issue, until the pr is merged
